h5py/h5d.pyx

```diff
@@ -430,3 +431,65 @@ cdef class DatasetID(ObjectID):
                 efree(offset)
                 if space_id:
                     H5Sclose(space_id)
+
+    IF HDF5_VERSION >= (1, 10, 2):
+
+        def read_direct_chunk(self, offsets, filter_mask=0xFFFF, PropID dxpl=None):
```

Comment:
For parity with `write_direct_chunk`, I think `filter_mask` should probably default to 0 (i.e. apply all enabled filters), rather than `0xFFFF` (i.e. skip all filters).

Reply:
My default expectation is to read the raw data, which is the case by disabling all the filters. Else, we have to inject this magic `0xFFFF`, which is IMO not very nice. Or we have to create on our side a new const var. Maybe i misunderstand something.

BTW i just check the HDF5 API `H5DOwrite_chunk`, and it do not use filter type `H5Z_filter_t`, i am pretty sure it was the case before! Or it was a mistake from `h5py`? Then `H5Z_filter_t filter_mask=H5Z_FILTER_NONE` should be rewritten to `uint_t filter_mask=0x0000`. Which far way much better.

In this case, if `H5Z_FILTER_NONE` is not used in the writer, i think it is fine to use a magic `0xFFFF` on application side.

BTW do you use a resource to follow the changes of the HDF5 API?

Reply:
Oh `H5DOwrite_chunk` from `api_functions.txt` uses `uint32_t`. Then it's fine. I will update the thing as you suggest. And patch `write_direct_chunk` on the same time.